File: Java/Easy/258-add-digits/add-digits.java
class Solution {
    public int addDigits(int num) {
        int n = num;

        while (n >= 10) {
            int sum = 0;
            while (n != 0) {
                sum += n % 10;
                n = n / 10;
            }
            n = sum;   // sum is assigned back to n for next round
        }
        return n;      // final single-digit result
    }
}

File: Python/Easy/258-add-digits/add-digits.cpp
class Solution {
public:
    int addDigits(int num) {
        while(num>9){
            int sum=0;
        while(num>0){
        sum+=num%10;
        num=num/10;
        }
          num=sum;  // Since sum was declared inside the while loop !
        }
        return num;
    }
};